File: Python/lab4/factory/ProductFactory.py
from factory.LaptopFactory import LaptopFactory
from factory.PcFactory import PcFactory
from factory.PhoneFactory import PhoneFactory
from interfaces.Factory import Factory

class ProductFactory():
    def create_factory(self, type_of_factory) -> Factory:
        if type_of_factory == 'Phone':
            return PhoneFactory()
        elif type_of_factory == "PC":
            return PcFactory()
        elif type_of_factory == "Laptop":
            return LaptopFactory()
        else:
            return None

File: Java/lab4/src/main/java/org/example/factory/ProductFactory.java
package org.example.factory;

import org.example.interfaces.Factory;
public class ProductFactory {
    public Factory createFactory(String typeFactory) {
        return switch (typeFactory) {
            case "Phone" -> new PhoneFactory();
            case "Laptop" -> new LaptopFactory();
            case "PC" -> new PcFactory();
            default -> null;
        };
    }
}
